doc/qos/tunnel_dscp_remapping.md

```diff
@@ -0,0 +1,258 @@
+
+Please be noted that below config is to remap traffic in queue 3 to queue 2, and traffic in queue 4 to queue 6.
+Before remapping to queue 4 and 6, both queues are required to be cleared. Hence the current `DSCP_TO_TC_MAP|AZURE` in [qos_config.j2](https://github.com/Azure/sonic-buildimage/blob/master/files/build_templates/qos_config.j2) is required to be updated to map DSCP `5` and `48` into other queues.
+* Table for decap
+
+    DSCP_TO_TC_MAP for mapping DSCP to TC
+
+    ```json
+    "DSCP_TO_TC_MAP": {
+        "AZURE_TUNNEL": {
+            "0": "1",
+            "1": "1",
+            "2": "2",
+            "3": "3",
+            "4": "4",
+            "5": "1",
+            "6": "6",
+            "7": "1",
+            "8": "0",
+            "9": "1"
+    }
+    ```
+
+    TC_TO_PRIORITY_GROUP_MAP for mappping TC to PG
+
+    ```json
+    "TC_TO_PRIORITY_GROUP_MAP": {
+        "AZURE_TUNNEL": {
+            "0": "0",
+            "1": "0",
+            "2": "0",
+            "3": "2",
+            "4": "6",
+            "5": "0",
+            "6": "0",
+            "7": "7"
+    }
+    ```
+
+* Table for encap
+
+    TC_TO_QUEUE_MAP for remapping queue
+
+    ```json
+    "TC_TO_QUEUE_MAP": {
+        "AZURE_TUNNEL": {
+            "0": "0",
+            "1": "1",
+            "2": "1",
+            "3": "2",
+            "4": "6",
+            "5": "5",
+            "6": "1",
+            "7": "7"
+    }
+    ``` 
+
+    TC_TO_DSCP_MAP for rewriting DSCP
+
+    ```json
+    "TC_TO_DSCP_MAP": {
+        "AZURE_TUNNEL": {
+            "0": "8",
+            "1": "0",
+            "2": "0",
+            "3": "2",
+            "4": "6",
+            "5": "46",
+            "6": "0",
+            "7": "48"
+    }
+    ```
+
+    To support the new table, a new YANG model `sonic-tc-dscp.yang` is required 
+#### 5.1.2 Update existing TUNNEL table
+1. Change `dscp_mode` from `uniform` to `pipe` for TC remapping
+2. Add TC remapping config if TC remapping is enabled
+
+```json
+    "TUNNEL": {
+        "MuxTunnel0": {
+            "dscp_mode": "pipe",
+            "dst_ip": "10.1.0.32",
+            "ecn_mode": "copy_from_outer",
+            "encap_ecn_mode": "standard",
+            "ttl_mode": "pipe",
+            "tunnel_type": "IPINIP",
+            "decap_dscp_to_tc_map": "[DSCP_TO_TC_MAP|AZURE_TUNNEL]",
+            "decap_tc_to_pg_map": "[TC_TO_PRIORITY_GROUP_MAP|AZURE_TUNNEL]",
+            "encap_tc_to_queue_map": "[TC_TO_QUEUE_MAP|AZURE_TUNNEL]",
+            "encap_tc_color_to_dscp_map": "[TC_TO_DSCP_MAP|AZURE_TUNNEL]"
+        }
+    }
+```
+
+#### 5.1.3 Define new field for extra lossless queues
+Two new fields are added to specify software or hardward PFC watchdog.
+
+* `pfc_wd_sw_enable` Specify the queue(s) to enable software PFC watchdog
+* `pfc_wd_hw_enable` Specify the queue(s) to enable hardware PFC watchdog.
```

Comment:
If only SW PFC watchdog is supported then 2,3,4,6 should be listed with "pfc_wd_sw_enable" ?

Reply:
It depends. We are not able to have 4 software watchdogs on some platforms. So we are attempting to implement hardware PFC watchdog. So, the final state might be
```
pfc_wd_sw_enable: '3,4'
pfc_wd_hw_enable: '2,6'
```
Of course, if only software watcdog is supported, and we can have 4 software watchdogs, all `2,3,4,6` should be in `pfc_wd_sw_enable`